File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_007)_Left_Rotate_Array_by_d_(Time_N).cpp
#include <iostream>
#include <cmath>
using namespace std;

void leftRotate(int arr[], int d, int n)
{
	int temp[d];
	for(int i = 0; i  < d; i++)
	{
		temp[i] = arr[i];
	}
	for(int i = d; i  < n; i++)
	{
		arr[i - d] = arr[i];
	}
	for(int i = 0; i  < d; i++)
	{
		arr[n - d + i] = temp[i];
	}
}

int main()
{
      int arr[] = {1, 2, 3, 4, 5}, n = 5, d = 2;
      cout<<"Before Left Rotation"<<endl;
       for(int i = 0; i < n; i++)
       {
       		cout<<arr[i]<<" ";
       }
       cout<<endl;
       leftRotate(arr, d, n);
       cout<<"After '2' Left Rotation"<<endl;
       for(int i = 0; i < n; i++)
       {
       		cout<<arr[i]<<" ";
       }
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_007)_Left_Rotate_Array_by_d_(Time_N).dart
void leftRotate(arr, int d, int n) {
  var temp = List.empty(growable: true)..length = d;

  for (int i = 0; i < d; i++) {
    temp[i] = arr[i];
  }
  for (int i = d; i < n; i++) {
    arr[i - d] = arr[i];
  }
  for (int i = 0; i < d; i++) {
    arr[n - d + i] = temp[i];
  }
}

void main() {
  var arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
  print('Before Left Rotate: ${arr}');
  leftRotate(arr, 3, arr.length);
  print('After Left Rotate: ${arr}');
}
